tuf/store/interfaces.go

```diff
@@ -12,7 +17,8 @@ type MetadataStore interface {
 
 // PublicKeyStore must be implemented by a key service
 type PublicKeyStore interface {
-	GetKey(role string) ([]byte, error)
+	GetKey(role string) (data.PublicKey, error)
+	RotateKey(role string, cs signed.CryptoService, roots ...data.PublicKey) (data.PublicKey, error)
```

Comment:
I agree that I wasn't super happy the arguments for `RotateKey` - perhaps a `KeyRotationHTTPStore` (or just `HTTPStore`) can take the `CryptoService`, etc., and can generate a key rotation request based on that?  Then the `RotateKey` parameters would just be the role as a string.

How do you feel about these functions returning `data.PublicKey` though instead of `[]byte`?  For metatdata especially, it makes sense for a remote store interface to return bytes, since we have to calculate the hash of those bytes.   But since no validation is done on the key bytes other than to make sure they're unmarshallable as keys, it seems anything that wants to use this interface ultimately cares about the keys, not the bytes themselves.  So it seems to make sense for the interface to return `data.PublicKey`?